orc/src/test/java/org/apache/iceberg/orc/TestORCSchemaUtil.java

```diff
@@ -0,0 +1,231 @@
+        required(22, "binaryCol", Types.BinaryType.get()),
+        required(23, "stringCol", Types.StringType.get()),
+        required(24, "decimalCol", Types.DecimalType.of(15, 3)),
+        required(25, "floatCol", Types.FloatType.get()),
+        optional(30, "dateCol", Types.DateType.get()),
+        required(32, "timeCol", Types.TimeType.get()),
+        required(34, "timestampCol", Types.TimestampType.withZone())
+    );
+    TypeDescription orcSchema = ORCSchemaUtil.convert(expectedSchema);
+    assertEquals(expectedSchema.asStruct(), ORCSchemaUtil.convert(orcSchema).asStruct());
+  }
+
+  @Test
+  public void testRoundtripConversionNested() {
+    Types.StructType leafStructType = Types.StructType.of(
+        optional(6, "leafLongCol", Types.LongType.get()),
+        optional(7, "leafBinaryCol", Types.BinaryType.get())
+    );
+    Types.StructType nestedStructType = Types.StructType.of(
+        optional(4, "longCol", Types.LongType.get()),
+        optional(5, "leafStructCol", leafStructType)
+    );
+    // all fields in expected iceberg schema will be optional since we don't have a column mapping
+    Schema expectedSchema = new Schema(
+        optional(1, "intCol", Types.IntegerType.get()),
+        optional(2, "longCol", Types.LongType.get()),
+        optional(3, "nestedStructCol", nestedStructType),
+        optional(8, "intCol3", Types.IntegerType.get()),
+        optional(9, "doubleCol", Types.DoubleType.get()),
+        required(10, "uuidCol", Types.UUIDType.get()),
+        optional(20, "booleanCol", Types.BooleanType.get()),
+        optional(21, "fixedCol", Types.FixedType.ofLength(4096)),
+        required(22, "binaryCol", Types.BinaryType.get()),
+        required(23, "stringCol", Types.StringType.get()),
+        required(24, "decimalCol", Types.DecimalType.of(15, 3)),
+        required(25, "floatCol", Types.FloatType.get()),
+        optional(30, "dateCol", Types.DateType.get()),
+        required(32, "timeCol", Types.TimeType.get()),
+        required(34, "timestampCol", Types.TimestampType.withZone())
+    );
+    TypeDescription orcSchema = ORCSchemaUtil.convert(expectedSchema);
+    assertEquals(expectedSchema.asStruct(), ORCSchemaUtil.convert(orcSchema).asStruct());
+  }
+
+  @Test
+  public void testSchemaEvolutionPrimitiveNoOp() {
+    Schema originalSchema = new Schema(
+        optional(1, "a", Types.IntegerType.get()),
+        optional(2, "b", Types.StringType.get())
+    );
+
+    // Original mapping (stored in ORC)
+    TypeDescription orcSchema = ORCSchemaUtil.convert(originalSchema);
+    assertEquals(2, orcSchema.getChildren().size());
+    assertEquals(1, orcSchema.findSubtype("a").getId());
+    assertEquals(TypeDescription.Category.INT, orcSchema.findSubtype("a").getCategory());
+    assertEquals(2, orcSchema.findSubtype("b").getId());
+    assertEquals(TypeDescription.Category.STRING, orcSchema.findSubtype("b").getCategory());
+  }
+
+  @Test
+  public void testSchemaEvolutionPrimitive() {
```

Comment:
It looks like these test cases are actually testing `buildOrcProjection` and not schema evolution. I think it would be more clear to put them in a test suite specific to `buildOrcProjection`.

Also, it may help to produce your evolved schema by using the `SchemaUpdate` class. Then you'd be able to write like this:

```java
Schema evolvedSchema = new SchemaUpdate(originalSchema, 2)
    .deleteColumn("a")
    .renameColumn("b", "a")
    .addColumn("c", DateType.get())
    .apply();
```

Reply:
I split these tests into their own test suite for build ORC projection. I looked into using `SchemaUpdate` however it is not a public class and seems to only be accessible via `Table` with the schema evolution API. I'm not sure I'd want to change the access modifier for this class nor create a Table for this test (seems overkill) but I'm open to suggestions. Thanks!

Reply:
I'd be fine with making it public. The API is in the iceberg-api module, so there aren't strong guarantees if you use public classes in core.